### PR TITLE
Add Github actions for npm checks

### PR DESCRIPTION
**Pull request by MSevey**
# PULL REQUEST

## Overview

Adding basic github actions for kernel repo. Since this repo is structured with many subpackages, I followed the webportal standard of creating an action per package to help manage. 

Also ran prettier on some of the packages that didn't have prettier added yet. 

## Checklist

- [ ] All git commits are signed. (REQUIRED)
- [ ] All new methods or updated methods have clear docstrings.
- [ ] Testing added or updated for new methods.
- [ ] Verified if any changes impact the WebPortal Health Checks.
- [ ] Appropriate documentation updated.
- [ ] Changelog file created.

## Issues Closed

Closes SKY-767

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: 2
+updates:
+  - package-ecosystem: npm
+    directory: "/"
+    schedule:
+      interval: monthly
+    open-pull-requests-limit: 10
```
Comment (kwypchlo) [inline .github/dependabot.yml:7]: will it work for all package.json in modules directories or do we have to define each one separately ?

**Review by kwypchlo — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (MSevey) [reply .github/dependabot.yml:7]: created issue https://linear.app/skynetlabs/issue/SKY-782/determine-how-dependabot-works-for-subpackage-packagejson-files

File: .github/workflows/ci_libkerneldev.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: "CI - libkerneldev"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    paths:
+      - lib/libkerneldev/**
```
Comment (m-cat) [inline .github/workflows/ci_libkerneldev.yml:8]:
```suggestion
      - libs/libkerneldev/**
```

File: .github/workflows/ci_libmodule.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: "CI - libmodule"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    paths:
+      - lib/libmodule/**
```
Comment (m-cat) [inline .github/workflows/ci_libmodule.yml:8]:
```suggestion
      - libs/libmodule/**
```

File: .github/workflows/ci_libskynet.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+name: "CI - libskynet"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    paths:
+      - lib/libskynet/**
```
Comment (m-cat) [inline .github/workflows/ci_libskynet.yml:8]:
```suggestion
      - libs/libskynet/**
```

**Review by m-cat — APPROVED** [review]

File: webapps/kernel-test-suite/src/pages/index.js [hunk 5]
```diff
@@ -75,7 +75,10 @@ function TestGetKernelVersion() {
 // 		kernel => test module ->
 // 		kernel ->
 // 	background -> bridge -> webpage
+<<<<<<< HEAD
 let kernelTestSuite = "AQCPJ9WRzMpKQHIsPo8no3XJpUydcDCjw7VJy8lG1MCZ3g";
+=======
+>>>>>>> main
```
Comment (kwypchlo) [inline webapps/kernel-test-suite/src/pages/index.js:81]: committed merge conflict

File: webapps/kernel-test-suite/src/pages/index.js [hunk 6]
```diff
@@ -902,5 +905,40 @@ const IndexPage = () => {
...
+			<TestCard name="TestMsgSpeedSequential5k" test={TestMsgSpeedSequential5k} turn={getTurn()} />
+			<TestCard name="TestModuleSpeedSeq20k" test={TestModuleSpeedSequential20k} turn={getTurn()} />
+			<TestCard name="TestMsgSpeedParallel5k" test={TestMsgSpeedParallel5k} turn={getTurn()} />
+			<TestCard name="TestModuleSpeedParallel20k" test={TestModuleSpeedParallel20k} turn={getTurn()} />
+			<TestCard name="TestModuleHasErrors" test={TestModuleHasErrors} turn={getTurn()} />
+			<TestCard name="TestHelperModuleHasErrors" test={TestHelperModuleHasErrors} turn={getTurn()} />
+		</main>
+	)
+}
+>>>>>>> main
```
Comment (kwypchlo) [inline webapps/kernel-test-suite/src/pages/index.js:942]: committed merge conflict

**Review by kwypchlo — CHANGES_REQUESTED** [review]

**Review by m-cat — APPROVED** [review]

**Review by kwypchlo — APPROVED** [review]

**Comment by MSevey** [comment]
resolve conflicts

**Pull request merged**